src/util/hacky.ts

```diff
@@ -93,138 +93,149 @@ export function getTotalOwedAccurate(steward: Steward): BigInt {
-        .times(steward.price(BigInt.fromI32(7)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(8))
-        .times(steward.price(BigInt.fromI32(8)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(9))
-        .times(steward.price(BigInt.fromI32(9)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(10))
-        .times(steward.price(BigInt.fromI32(10)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(11))
-        .times(steward.price(BigInt.fromI32(11)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(12))
-        .times(steward.price(BigInt.fromI32(12)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(13))
-        .times(steward.price(BigInt.fromI32(13)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(14))
-        .times(steward.price(BigInt.fromI32(14)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(15))
-        .times(steward.price(BigInt.fromI32(15)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(16))
-        .times(steward.price(BigInt.fromI32(16)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(17))
-        .times(steward.price(BigInt.fromI32(17)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(18))
-        .times(steward.price(BigInt.fromI32(18)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(19))
-        .times(steward.price(BigInt.fromI32(19)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(20))
-        .times(steward.price(BigInt.fromI32(20)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(21))
-        .times(steward.price(BigInt.fromI32(21)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(22))
-        .times(steward.price(BigInt.fromI32(22)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(23))
-        .times(steward.price(BigInt.fromI32(23)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(24))
-        .times(steward.price(BigInt.fromI32(24)))
-    )
-    .plus(
-      steward
-        .patronageNumerator(BigInt.fromI32(25))
-        .times(steward.price(BigInt.fromI32(25)))
-    )
-    .plus(
-      BigInt.fromI32(300)
-        .times(BigInt.fromI32(1000000000))
-        .times(steward.price(BigInt.fromI32(42)))
-    );
+export function getTotalTokenCostScaledNumerator(
+  steward: Steward,
+  delta: BigInt
+): BigInt {
+  let globalState = Global.load("1");
+  let currentVersion = globalState.version;
+
+  // execure correct function based on on version.
+  if (currentVersion.ge(BigInt.fromI32(3))) {
+    return globalState.totalTokenCostScaledNumeratorAccurate.plus(delta);
```

Comment:
If we are in version 3, we simply return the globalState.totalTokenCostScaledNumeratorAccurate value plus the delta.

TotalTokenCostScaledNumerator = TK1patronageNumerator * TK1price +    ….     + TKnpatronageNumerator * TKnprice

Change in  TotalTokenCostScaledNumerator = TotalTokenCostScaledNumerator + delta

Delta = (TKnpatronageNumerator * TKnNewprice) - (TKnpatronageNumerator * TKnOldprice) Delta = TKnpatronageNumerator * (TKnNewprice-TKnOldprice)

Reply:
Call the function with delta 0 and value will stay the same